File: 1089-duplicate-zeros/1089-duplicate-zeros.js
/**
 * @param {number[]} arr
 * @return {void} Do not return anything, modify arr in-place instead.
 */
var duplicateZeros = function(arr) {
    
    const total = []
    for(let x of arr){
        if(x == 0){
            total.push(x)
            total.push(x)
        }else{
            total.push(x)
        }
    }
    
    for(let i = 0; i<arr.length; i++){
        arr[i] = total[i]
    }
    
};

File: 1089-duplicate-zeros/1089-duplicate-zeros.java
class Solution {
    public void duplicateZeros(int[] arr) {
        ArrayList<Integer> res = new ArrayList<Integer>();
        for(int n: arr){
            if(n == 0){
                res.add(n);
                res.add(n);
            }else{
                res.add(n);
            }
        }
        
        for(int i = 0; i<arr.length; i++){
            arr[i] = res.get(i);
        }
    }
}